troubleshooting/macbook-troubleshooting.md

```diff
@@ -139,3 +139,182 @@ Please see [this thread o the qubes-devel mailing list][macbook-air-2012-5-1].
+   ~~~
+   crsutil disable
+   ~~~
+   * reboot and you will see some icons
+   * choose Boot EFI\BOOT\xen.efi from ANACONDA
+   * after a will the graphical installer is up and running, with keyboard and touchpad working
+
+### 3. Installation
+
+*  As a general rule, keep the default values proposed during installation: you will change them later on
+*  Keep English, as language, locale
+*  My macbook has a US keyboard, so I cannot say what happens if you change keyboard layout
+*  DO NOT CHANGE the timezone, because it will trigger the wifi card, leading to a system freeze
+*  Choose the "installation destination": do not change anything and press DONE button
+*  Insert your password for Full Disk Encryption
+*  If you do not already have free space on internal SSD disk, you will be prompted to reclaim some space: 
+*  If you shrunk OSX partition, disk utility left an empy partition: delete useless partition (eg: if you shrunk OSX parition, diskutil  created an empty partition)
+*  Press on "reclaim space"
+*  Press on "begin installation"
+*  create your user and password
+*  after a while, installation completes
+*  DO NOT press "Reboot button"
+
+Qubes OS is now installed, but you cannot boot it due to some issues, with bootloader configuration and wifi card.
+You cannot Qubes boot using EFI/qubes/xen.efi because XEN bootloader configuration is broken. 
+You cannot even Qubes without XEN support, using GRUB2, because its configuration is broken too.
+
+Let's fix it manually, switch to console, pressing Fn+CTRL+ALT+F2
+
+### 4. Fix GRUB configuration
+
+You can skip this section, but I found it very useful - during troubleshooting - to have a rescue system at hand. I could boot Qubes, without XEN support
+
+Grub configuration file is using some wrong commands, which are not compatible with grub2-efi
+~~~
+chroot /mnt/sysimage
+sed -i.bak -e "s/multiboot/chainloader/" -e "s/module.*--nounzip/initrdefi/" -e "s/module/linuxefi/" /etc/grub.d/20_linux_efi
+exit
+~~~
+
+Now, despite XEN configuration is still broken, you have a rescue system booting vmlinux from rEFInd screen.
+TBV1: chainloading XEN does not work, unless you specify the right disk prefix, eg: (hd1,gpt4)
+TBV2: grub.cfg set the wrong disk in "set root" command
+TBV3: in case you reach grub shell, you can 
+~~~
+ls
+~~~
+and also reload config file and change it manualy before booting
+~~~
+configfile /EFI/qubes/grub.cfg
+~~~
+then press "e", edit grub cfg and boot pressing Fn+F10
+
+
+### 5. Fix bootloader
+
+*  Fix grub2 configuration, which uses wrong command for EFI boot
+*  analyzing /mnt/sysimage/var/log/anaconda/program.log I found the faulty commands issues by Anaconda installer
+~~~
+chrooot /mnt/sysimage
+~~~
+* edit /boot/efi/EFI/qubes/xen.cfg file putting the following content
+~~~
+[global]
+default=4.4.14-11.pvops.qubes.x868_64
+
+[4.4.14-11.pvops.qubes.x868_64]
+options=loglvl=all
```

Comment:
Add `dom0_mem=min:1024M dom0_mem=max:4096M` to `options` (it's default in 3.2, was missing in 3.1).